src/main/kotlin/com/ekino/oss/gradle/plugin/java/JavaPlugin.kt

```diff
@@ -32,15 +24,21 @@ class JavaPlugin : Plugin<Project> {
   override fun apply(project: Project) {
     with(project) {
 
+      extensions.create(EXTENSION_NAME, JavaPluginConfiguration::class.java)
+
       // plugins
       apply<org.gradle.api.plugins.JavaPlugin>()
       apply<TestSetsPlugin>()
       val sourceSets = the<SourceSetContainer>()
       apply<MavenPublishPlugin>()
 
       // properties
-      setProperty("sourceCompatibility", JavaVersion.VERSION_11)
-      setProperty("targetCompatibility", JavaVersion.VERSION_11)
+      afterEvaluate {
+        with(javaPluginConfiguration()) {
```

Comment:
it's debatable but to remove `javaPluginConfiguration()` you can do:
```kt
val javaPluginExtension = extensions.create<JavaPluginConfiguration>(EXTENSION_NAME)
...
afterEvaluate {
  with(javaPluginExtension) {
    setProperty("sourceCompatibility", sourceCompatibility)
    setProperty("targetCompatibility", targetCompatibility)
  }
}
```

Reply:
You're right and the extension can be removed too.